primitives/runtime/src/transaction_validity.rs

```diff
@@ -59,6 +59,13 @@ pub enum InvalidTransaction {
 	/// A transaction with a mandatory dispatch. This is invalid; only inherent extrinsics are
 	/// allowed to have mandatory dispatches.
 	MandatoryDispatch,
+	/// No validator has been able to fully verify this transaction.
+	///
+	/// The transactions are disallowed from being included by default. Some validator
+	/// (`SignedExtension`) needs to whitelist them first. This error usually means that either you
```

Comment:
For signed transactions, `SignedExtension` works by default-pass (blacklisting), not whitelisting.

Reply:
That's only partially true, since the `ValidTransaction` produced will have no `provides` tags, which will be rejected by the transaction pool anyway.

Now `CheckedExtrinsic` will explicitly return `InvalidTransaction` error if no `SignedExtension` provides any tags in `validate` call. I also unified that with `pre_dispatch` now, so basically it's a whitelist now (in a sense that you need at least one `SignedExtension` to provide some tags, don't need to whitelist specific calls or anything).